src/core/modules/impl/handler/JsHandlerApi.ts

```diff
@@ -75,18 +110,35 @@ export class JsHandlerApi<State> extends AbstractContractHandler<State> {
       }
     } finally {
       await this.swGlobal.kv.close();
-      if (timeoutId !== null) {
-        // it is important to clear the timeout promise
-        // - promise.race won't "cancel" it automatically if the "handler" promise "wins"
-        // - and this would ofc. cause a waste in cpu cycles
-        // (+ Jest complains about async operations not being stopped properly).
-        clearTimeout(timeoutId);
-      }
     }
   }
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   initState(state: State): void {
-    // nth to do in this impl...
+
+  }
+
+  async maybeCallStateConstructor<Input>(initialState: State, executionContext: ExecutionContext<State>): Promise<State> {
+    if (this.contractDefinition.manifest?.evaluationOptions.useConstructor) {
+      const interaction = { input: { function: INIT_FUNC_NAME, args: initialState } as Input, caller: this.contractDefinition.owner };
+      // TODO: we have to use some default sort key like 0,0,0,0
+      const interactionData: InteractionData<Input> = { interaction, interactionTx: { ...this.contractDefinition.contractTx, sortKey: '0.0.0.0' } };
```

Comment:
Hack with `sortKey` to make KV works, however it need fix now.
I see two options:
1) don't enable KV and internal writes in constructor (bad user experience)
2) generate some hacky sortKey which will be always first 
3) make constructor sperate transaction (need additional signing)
4) dont' know

Reply:
Hmmmmmm, internal writes are impossible to implement in constructor if we don't produce new interaction (I think we don't wont to do this).
So new options:
Change implementation which produce new interaction `__init` (a little bit shady, more code)
Leave as is, with working KV in `__init` function (constructor), but without support for KV (only in __init function)